odoo/addons/test_new_api/tests/test_new_fields.py

```diff
@@ -3056,6 +3056,32 @@ def test_cache_key_invalidation(self):
         self.assertEqual(record.env.company, company1)
         self.assertEqual(record.foo, False)
 
+    def test_field_set_prefetch(self):
+        source_records = self.env['test_new_api.prefetch.modified'].create([
+            {'lines_ids': [Command.create({})]},
+            {'lines_ids': [Command.create({})]},
+            {'lines_ids': [Command.create({})]},
+        ])
+
+        # We ensure in this test that the prefetch set is preserved when using Field.__set__()
+        # which calls BaseModel.write(). The prefetch set is important for write() to ensure
+        # that BaseModel.modified() can batch the fetching. In that case:
+        # changing the `value` of the corresponding `test_new_api.prefetch.modified` will add
+        # test_new_api.prefetch.modified.line.value to recompute through the one2many inverse
+        # test_new_api.multi.lines_ids which can be get in batch with the prefetch set
+
+        self.env.invalidate_all()
+        # One for the modified; One UPDATE in source; One UPDATE on lines_ids
+        with self.assertQueryCount(3):
+            for source_record, new_value in zip(source_records, [1, 2, 3]):
```

Comment:
Can you change the size of the list so it's not confused with 3 in query count?
```suggestion
            for source_record, new_value in zip(source_records, range(5)):
```